contracts/pcv/maple/MaplePCVDeposit.sol

```diff
@@ -0,0 +1,118 @@
+    IMplRewards public immutable mplRewards;
+
+    /// @notice reference to the underlying token
+    IERC20 public immutable token;
+
+    /// @notice scaling factor
+    uint256 public constant scalingFactor = 1e12;
+
+    /// @notice fetch underlying asset by calling pool and getting liquidity asset
+    /// @param _core reference to the Core contract
+    /// @param _pool reference to the Maple Pool contract
+    constructor(
+        address _core,
+        address _pool,
+        address _mplRewards
+    ) CoreRef(_core) {
+        pool = IPool(_pool);
+        token = IERC20(IPool(_pool).liquidityAsset());
+        mplRewards = IMplRewards(_mplRewards);
+    }
+
+    /// @notice return the amount of funds this contract owns in Maple FDT's
+    function balance() public view override returns (uint256) {
+        return pool.balanceOf(address(this)) / scalingFactor;
+    }
+
+    /// @notice return the underlying token denomination for this deposit
+    function balanceReportedIn() external view returns (address) {
+        return address(token);
+    }
+
+    /// @notice deposit PCV into Maple.
+    /// all deposits are subject to a minimum 90 day lockup,
+    /// no op if 0 token balance
+    /// deposits are then immediately staked to accrue MPL rewards
+    /// only pcv controller can deposit, as this contract would be vulnerable
+    /// to donation / griefing attacks if anyone could call deposit and extend lockup time
+    function deposit() external onlyPCVController {
+        uint256 amount = IERC20(token).balanceOf(address(this));
+        if (amount == 0) {
+            /// no op to prevent wasted gas
+            return;
+        }
+
+        /// pool deposit
+        token.approve(address(pool), amount);
+        pool.deposit(amount);
+
+        /// token withdraw
+        uint256 scaledDepositAmount = amount * scalingFactor;
+        pool.increaseCustodyAllowance(address(mplRewards), scaledDepositAmount);
+        mplRewards.stake(scaledDepositAmount);
+
+        emit Deposit(msg.sender, amount);
+    }
+
+    /// @notice function to start the cooldown process to withdraw
+    function signalIntentToWithdraw() external onlyPCVController {
+        pool.intendToWithdraw();
+    }
+
+    /// @notice function to cancel a withdraw
+    /// should only be used to allow a transfer when doing a withdrawERC20 call
+    function cancelWithdraw() external onlyPCVController {
+        pool.cancelWithdraw();
+    }
+
+    /// @notice withdraw PCV from Maple, only callable by PCV controller
+    /// @param to destination after funds are withdrawn from venue
+    /// @param amount of PCV to withdraw from the venue
+    function withdraw(address to, uint256 amount)
+        external
+        override
+        onlyPCVController
+    {
+        uint256 scaledWithdrawAmount = amount * scalingFactor;
+
+        mplRewards.getReward(); /// get MPL rewards
+        mplRewards.withdraw(scaledWithdrawAmount); /// decreases allowance
```

Comment:
make a note that this will withdraw more than you ask for if any interest has accrued due to their implementation. this is behavior we expect and understand.